pkg/reconciler/inmemorychannel/controller/inmemorychannel_test.go

```diff
@@ -110,11 +110,61 @@ func TestAllCases(t *testing.T) {
 			WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
 				Object: reconciletesting.NewInMemoryChannel(imcName, testNS,
 					reconciletesting.WithInitInMemoryChannelConditions,
-					reconciletesting.WithInMemoryChannelDeploymentNotReady("DispatcherDeploymentDoesNotExist", "Dispatcher Deployment does not exist")),
+					reconciletesting.WithInMemoryChannelDeploymentFailed("DispatcherDeploymentDoesNotExist", "Dispatcher Deployment does not exist")),
```

Comment:
I checked the reconcilers. We use "false" when resource doesn't exist and we use "unknown" when we fail to get resource. This also matches the conclusion of discussion in https://github.com/knative/eventing/issues/2187#issuecomment-562283838. Maybe I missed some reconcilers. Could you help share an example of reconciler which calls this unknown?

Reply:
You're right. I was thinking that the IMC reconciler creates the dispatcher deployment during reconcile. In that case it seemed like a DNE error would be Unknown, because we'd expect the reconciler to fix it next time by creating the deployment.

Now I realize that I was mistaken: the dispatcher deployment is created during IMC install, not during reconcile. If the deployment doesn't exist, it should be a failed condition because the reconciler can't fix it.

Thanks for checking my assumptions @capri-xiyue!